swingset/src/main/java/com/nqadmin/swingset/SSBaseComboBox.java

```diff
@@ -111,30 +112,73 @@
 	{
 		private static final long serialVersionUID = -3131533966245488092L;
 
-		/** {@inheritDoc} */
+		/**
+		 * For JComboBox ActionListener and ItemListener are very similar.
+		 * 
+		 * {@inheritDoc} */
 		@Override
 		public void actionPerformed(final ActionEvent ae) {
 
-			removeRowSetListener();
-
-			M mapping = getSelectedMapping();
-
-			if (mapping == null) {
-				setBoundColumnText(null);
-				logger.debug(() -> String.format("%s: Setting to null.", getColumnForLog()));
-			} else {
-				setBoundColumnText(String.valueOf(mapping));
-				logger.debug(String.format("%s: Setting to %s.",  getColumnForLog(), mapping));
+			// If this is a combo navigator, SSSyncManager will have it's own listeners.
+			// This is just for keeping a bound column in sync.
+			//
+			// Could be combined with next block, but keeping them separate for debugging.
+			// May be able to eliminate actionListenerNoUpdate when GlazedLists fully supports STRICT/CONTAINS
+			if (getBoundColumnName() == null) {
+				logger.debug("{}: Action Listener returning. No bound column.", () -> getColumnForLog());
+				return;
 			}
-
-			addRowSetListener();
+			
+			// **** GL STRICT/CONTAINS ****
+			//
+			// Could be combined with prior block, but keeping them separate for debugging.
+			// May be able to eliminate actionListenerNoUpdate when GlazedLists fully supports STRICT/CONTAINS
+			//
+			// actionListenerNoUpdate will be set to true in setSelectedItem() when the user enters a non-matching string.
+			// It is also in the 'else' for unforeseen outcomes. The user is notified with a popup in both cases.
+			// 
+			// actionListenerNoUpdate is likely STRICT/CONTAINS workaround specific. Presumably if GL fully supported it,
+			// we'd only ever have null or an SSListItem passed to setSelectedItem(). Will have to put some thought into
+			// when we'd expect null (insert row, some escaping from/clearing of the combo editor)? Maybe we'd only ever
+			// have null for the insert row OR getModel().getSize()==0 ?
+			if (actionListenerNoUpdate) {
+				logger.debug("{}: Action Listener returning. actionListenerNoUpdate set to TRUE.", () -> getColumnForLog());
+				return;
+			}
+			
+			// EXTRACT SELECTED ITEM
+			Object selectedItem = getSelectedItem();
+			logger.debug("{}: ACTION LISTENER: getSelectedItem() has '{}'.", () -> getColumnForLog(), () -> selectedItem);
+			
+			// **** GL STRICT/CONTAINS ****
+			//
+			// IF SELECTED ITEM IS NOT AN SSLISTITEM, CHECK THE glGlitchItem
+			// 
+			// This seems to be a strange timing issue. The logs indicate that setSelectedItem() below has encountered 
+			// SCENARIO 1-B or 1-C and has made a call to super.setSelectedItem() to select the correct/updated SSListItem, but
+			// selectedItem here still has a String. glGlitchItem has the correct/updated SSListItem.
+			//
+			// Making a 2nd call to super.setSelectedItem(glGlitchItem) appears to resolve the issue and does not trigger an
+			// additional ActionListener event.
```

Comment:
This gave me fits, but  apparently the Action Listener gets out of sync with setSelectedItem(). It makes me wonder if things aren't being delivered to the logger in order.

Example 3. Type "am",<Enter> into Part and we get this in the logs:
15:30:11.070 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.SSBaseComboBox.setSelectedItem() - [PART_ID]: Null due to GlazedLists glitch so selecting first item in model. Selected item after super.setSelectedItem()={Cam,5}
15:30:11.362 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.SSBaseComboBox.actionPerformed() - [PART_ID]: ACTION LISTENER: getSelectedItem() has 'am'.
15:30:11.366 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.SSBaseComboBox.actionPerformed() - [PART_ID]:  -- About to call super.setSelectedItem({Cam,5}) {}.
15:30:11.371 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.SSBaseComboBox.actionPerformed() - [PART_ID]:  -- getSelectedItem() now returns {Cam,5}.
15:30:11.373 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.SSBaseComboBox.actionPerformed() - [PART_ID]: About to update RowSet with {Cam,5}.
15:30:11.374 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.datasources.RowSetOps.updateColumnText() - [PART_ID]. Update to: 5. Allow null? [false]
15:30:11.378 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.utils.SSComponentInterface.setBoundColumnText() - [PART_ID]: 5
15:30:11.379 [AWT-EventQueue-0] DEBUG com.nqadmin.swingset.SSBaseComboBox.updateRowset() - [PART_ID]: Setting column to 5.